e2e/utils/test_cases.js

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// See reference: https://support.smartbear.com/tm4j-cloud/api-docs/
+
+const axios = require('axios');
+
+const {getAllTests} = require('./report');
+
+const status = {
+    passed: 'Pass',
+    failed: 'Fail',
+    pending: 'Pending',
+    skipped: 'Skip',
+};
+
+const environment = {
+    chrome: 'Chrome',
+    firefox: 'Firefox',
+};
+
+function getStepStateResult(steps = []) {
+    return steps.reduce((acc, item) => {
+        if (acc[item.state]) {
+            acc[item.state] += 1;
+        } else {
+            acc[item.state] = 1;
+        }
+
+        return acc;
+    }, {});
+}
+
+function getStepStateSummary(steps = []) {
+    const result = getStepStateResult(steps);
+
+    let summary = '';
+    Object.entries(result).forEach(([key, value], index) => {
+        if (index) {
+            summary += ',';
+        }
+        summary += `${value} ${key}`;
+    });
+
+    return summary;
+}
+
+function getTM4JTestCases(report) {
+    const allTests = getAllTests(report.results);
+
+    const re = /(MM-T)\w+/g;
```

Comment:
Maybe change the regex to `/^(MM-T)\w+/g;` to make it starts with ticket number since the item title split for the key below assumes it is

Reply:
Good catch! I was switching back and forth on this earlier while testing some cases.